aws-ssm-maintenancewindowtask/src/main/java/software/amazon/ssm/maintenancewindowtask/CreateHandler.java

```diff
@@ -1,27 +1,79 @@
 package software.amazon.ssm.maintenancewindowtask;
 
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.RegisterTaskWithMaintenanceWindowRequest;
+import software.amazon.awssdk.services.ssm.model.RegisterTaskWithMaintenanceWindowResponse;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
 import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
 import software.amazon.cloudformation.proxy.Logger;
-import software.amazon.cloudformation.proxy.ProgressEvent;
 import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
 import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.ssm.maintenancewindowtask.translator.ExceptionTranslator;
+import software.amazon.ssm.maintenancewindowtask.translator.request.RegisterTaskWithMaintenanceWindowTranslator;
+import software.amazon.ssm.maintenancewindowtask.util.ClientBuilder;
+
 
 public class CreateHandler extends BaseHandler<CallbackContext> {
 
+    private static final SsmClient SSM_CLIENT = ClientBuilder.getClient();
+    private final RegisterTaskWithMaintenanceWindowTranslator registerTaskWithMaintenanceWindowTranslator;
+    private final ExceptionTranslator exceptionTranslator;
+
+    CreateHandler() {
+        this.registerTaskWithMaintenanceWindowTranslator = new RegisterTaskWithMaintenanceWindowTranslator();
+        this.exceptionTranslator = new ExceptionTranslator();
+    }
+
+    /**
+     * Used for unit tests.
+     *
+     * @param registerTaskWithMaintenanceWindowTranslator Translates ResourceModel objects into RegisterTaskWithMaintenanceWindow requests.
+     * @param exceptionTranslator Used for translating service model exceptions..
+     */
+    CreateHandler(final RegisterTaskWithMaintenanceWindowTranslator registerTaskWithMaintenanceWindowTranslator, final ExceptionTranslator exceptionTranslator) {
+        this.registerTaskWithMaintenanceWindowTranslator = registerTaskWithMaintenanceWindowTranslator;
+        this.exceptionTranslator = exceptionTranslator;
+    }
+
     @Override
     public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        logger.log(String.format("Processing CreateHandler request %s", request));
```

Comment:
I think we do have customer sensitive data in the register task request, can you check on that and make sure we don't log it.

Reply:
Yes you are right. All the task parameter details are sensitive data based on https://code.amazon.com/packages/MaintenanceWindowCommon/blobs/mainline/--/src/com/amazonaws/maintenancewindow/datastore/target/WindowTargetRecord.java
https://code.amazon.com/packages/MaintenanceWindowCommon/blobs/mainline/--/src/com/amazonaws/maintenancewindow/datastore/task/WindowTaskRecord.java. I'm working on a string converter right now to log only the insensitive data and I was planning to submit that in the next pr. Would it be better if I submit that in this pr instead?